ultralytics/data/utils.py

```diff
@@ -47,13 +47,88 @@ def img2label_paths(img_paths):
     return [sb.join(x.rsplit(sa, 1)).rsplit(".", 1)[0] + ".txt" for x in img_paths]
 
 
+def check_file_speeds(files, threshold_ms=10, max_files=5, prefix=""):
+    """
+    Check dataset file access speed and provide performance feedback.
+
+    This function tests the access speed of dataset files by measuring ping (stat call) time and read speed.
+    It samples up to 5 files from the provided list and warns if access times exceed the threshold.
+
+    Args:
+        files (list): List of file paths to check for access speed.
+        threshold_ms (float, optional): Threshold in milliseconds for ping time warnings.
+        max_files (int, optional): The maximum number of files to check.
+        prefix (str, optional): Prefix string to add to log messages.
+
+    Examples:
+        >>> from pathlib import Path
+        >>> image_files = list(Path("dataset/images").glob("*.jpg"))
+        >>> check_file_speeds(image_files, threshold_ms=15)
+    """
+    if not files or len(files) == 0:
+        LOGGER.warning(f"{prefix}WARNING ⚠️ Image speed checks: No files to check")
+        return
+
+    # Sample files (max 5)
+    files = random.sample(files, min(max_files, len(files)))
+
+    # Test ping (stat time)
+    ping_times = []
+    file_sizes = []
+    read_speeds = []
+
+    for f in files:
+        try:
+            # Measure ping (stat call)
+            start = time.time()
+            file_size = os.stat(f).st_size
+            ping_times.append((time.time() - start) * 1000)  # ms
+            file_sizes.append(file_size)
+
+            # Measure read speed
+            start = time.time()
+            with open(f, "rb") as file_obj:
+                _ = file_obj.read()
+            read_time = time.time() - start
+            if read_time > 0:  # Avoid division by zero
```

Comment:
How about measure the time to read all the files and then just average over it? something like this? 
```suggestion
start = time.perf_counter()
for f in files:
   with open(f, "rb") as file_obj:
       _ = file_obj.read()
read_time = (time.perf_counter() - start)/len(files)
'''

Reply:
Thought about this but then we'd have no +/- uncertanties, so leaving as is.